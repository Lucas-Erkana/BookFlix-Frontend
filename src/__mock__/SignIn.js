import React, { useState } from 'react';
import axios from 'axios';
import { Link, BrowserRouter } from 'react-router-dom';
import url from '../../redux/url';

const SignIn = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error] = useState(true);

  const handleLogin = (e) => {
    e.preventDefault();
    axios
      .post(`${url}login`, {
        user: {
          email,
          password,
        },
      })
      .then((response) => {
        localStorage.setItem('token', response.headers.authorization);
        const { role, id, full_name } = response.data.data;
        localStorage.setItem('role', role);
        localStorage.setItem('userId', id);
        localStorage.setItem('full_name', full_name);
        navigate('/');
      })
      .catch((error) => {
        console.error(error);
      });
  };

  return (
    <BrowserRouter>
      <div className="col-md col-sm-12 col-xs-12 container-main d-flex flex-row align-items-center login p-0">
        <div className="col-md-12 d-flex flex-column align-items-center session-overlay justify-content-center">
          <div className="input-box col-md d-flex flex-column align-items-center justify-content-center">
            <div className="intro">
              <span className="line" />
              <h4 className="intro__title">Welcome to HandyMate</h4>
              <p className="intro__text">Login to get started!</p>
            </div>
            <form onSubmit={handleLogin} className="sign-form">
              <div className="input-field">
                <input
                  className="sign-form__input"
                  type="email"
                  placeholder="Email"
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                />
              </div>
              <div className="input-field">
                <input
                  className="sign-form__input"
                  type="password"
                  placeholder="Password"
                  value={password}
                  onChange={(e) => setPassword(e.target.value)}
                />
              </div>
              <button
                type="submit"
                className="btn btn-primary rounded-pill px-4 py-2 text-light"
              >
                Login
              </button>
            </form>
            <div className="sign-in mt-4">
              <p>
                Do not have an account?&nbsp;
                <Link to="/signup" className="session-link-btn">Sign Up</Link>
              </p>
              {error === false && <p>Please enter valid username and password</p>}
            </div>
          </div>
        </div>
      </div>
    </BrowserRouter>
  );
};

export default SignIn;
