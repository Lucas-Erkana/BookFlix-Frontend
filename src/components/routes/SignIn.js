/* eslint-disable react/jsx-one-expression-per-line */
import React, { useState } from 'react';
import axios from 'axios';
import { useNavigate, Link } from 'react-router-dom';
import { FaEye, FaEyeSlash } from 'react-icons/fa';

const SignIn = () => {
  const navigate = useNavigate();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [errorMessage, setErrorMessage] = useState(null); // Updated from error to errorMessage
  const [showPassword, setShowPassword] = useState(false);

  const handleLogin = (e) => {
    e.preventDefault();
    axios
      .post('https://book-flix.onrender.com/login', {
        user: {
          email,
          password,
        },
      })
      .then((response) => {
        localStorage.setItem('token', response.headers.authorization);
        // eslint-disable-next-line camelcase
        const { role, id, full_name } = response.data.data;
        localStorage.setItem('role', role);
        localStorage.setItem('userId', id);
        localStorage.setItem('full_name', full_name);
        navigate('/');
      })
      .catch(() => { // Updated catch block
        setErrorMessage('Incorrect email or password');
      });
  };

  return (
    <div className="col-md col-sm-12 col-xs-12 container-main d-flex flex-row align-items-center login p-0">
      <div className="col-md-12 d-flex flex-column align-items-center session-overlay justify-content-center">
        <div className="input-box col-md d-flex flex-column align-items-center justify-content-center">
          <div className="intro">
            <span className="line" />
            <h4 className="intro__title">
              Welcome to <span className="book">Book</span>Flix
            </h4>
            <p className="intro__text">Login to get started!</p>
          </div>
          <form onSubmit={handleLogin} className="sign-form">
            <div className="input-field">
              <input
                className="sign-form__input"
                type="email"
                placeholder="Email"
                value={email}
                onChange={(e) => setEmail(e.target.value)}
              />
            </div>
            <div className="input-field">
              <div className="password-wrapper">
                <input
                  className="sign-form__input"
                  type={showPassword ? 'text' : 'password'}
                  placeholder="Password"
                  value={password}
                  onChange={(e) => setPassword(e.target.value)}
                />
                {showPassword ? (
                  <FaEyeSlash
                    onClick={() => setShowPassword(false)}
                    className="password-icon"
                  />
                ) : (
                  <FaEye
                    onClick={() => setShowPassword(true)}
                    className="password-icon"
                  />
                )}
              </div>
            </div>
            <button
              type="submit"
              className="btn btn-primary rounded-pill px-4 py-2 text-light"
            >
              Login
            </button>
          </form>
          <div className="sign-in mt-4">
            <p>
              Do not have an account?&nbsp;
              <Link to="/signup" className="session-link-btn">
                Sign Up
              </Link>
            </p>
            {errorMessage && <p id="show-error" className="error-message">{errorMessage}</p>}
          </div>
        </div>
      </div>
    </div>
  );
};

export default SignIn;
