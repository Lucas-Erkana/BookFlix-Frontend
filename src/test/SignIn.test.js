import { render, cleanup } from '@testing-library/react';
import SignIn from '../__mock__/SignIn';

afterEach(() => {
  cleanup();
});

describe('SignIn', () => {
  it('SignIn renders correctly', () => {
    const signin = render(<SignIn />);
    expect(signin).toMatchSnapshot();
  });
});
