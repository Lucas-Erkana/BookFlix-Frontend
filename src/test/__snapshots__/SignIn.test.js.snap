// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn SignIn renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-md col-sm-12 col-xs-12 container-main d-flex flex-row align-items-center login p-0"
      >
        <div
          class="col-md-12 d-flex flex-column align-items-center session-overlay justify-content-center"
        >
          <div
            class="input-box col-md d-flex flex-column align-items-center justify-content-center"
          >
            <div
              class="intro"
            >
              <span
                class="line"
              />
              <h4
                class="intro__title"
              >
                Welcome to HandyMate
              </h4>
              <p
                class="intro__text"
              >
                Login to get started!
              </p>
            </div>
            <form
              class="sign-form"
            >
              <div
                class="input-field"
              >
                <input
                  class="sign-form__input"
                  placeholder="Email"
                  type="email"
                  value=""
                />
              </div>
              <div
                class="input-field"
              >
                <input
                  class="sign-form__input"
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
              <button
                class="btn btn-primary rounded-pill px-4 py-2 text-light"
                type="submit"
              >
                Login
              </button>
            </form>
            <div
              class="sign-in mt-4"
            >
              <p>
                Do not have an account? 
                <a
                  class="session-link-btn"
                  href="/signup"
                >
                  Sign Up
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-md col-sm-12 col-xs-12 container-main d-flex flex-row align-items-center login p-0"
    >
      <div
        class="col-md-12 d-flex flex-column align-items-center session-overlay justify-content-center"
      >
        <div
          class="input-box col-md d-flex flex-column align-items-center justify-content-center"
        >
          <div
            class="intro"
          >
            <span
              class="line"
            />
            <h4
              class="intro__title"
            >
              Welcome to HandyMate
            </h4>
            <p
              class="intro__text"
            >
              Login to get started!
            </p>
          </div>
          <form
            class="sign-form"
          >
            <div
              class="input-field"
            >
              <input
                class="sign-form__input"
                placeholder="Email"
                type="email"
                value=""
              />
            </div>
            <div
              class="input-field"
            >
              <input
                class="sign-form__input"
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-primary rounded-pill px-4 py-2 text-light"
              type="submit"
            >
              Login
            </button>
          </form>
          <div
            class="sign-in mt-4"
          >
            <p>
              Do not have an account? 
              <a
                class="session-link-btn"
                href="/signup"
              >
                Sign Up
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
